### PR TITLE
React: updates to Add Datasets

### PR DESCRIPTION
**Pull request by cklamann**
* attach notes to dataset, not participant model in _bulk endpoint
* use defaults from session storage (if they exist) as initial search state
* create fresh defaults on form submission
* make notes field free text (though still technically autocomplete)
* update value of non-enum fields on blur
* add empty string default options where appropriate to suppress console warnings

### CODE REVIEW

**Comment by kevinlul** [comment]
Does this address the problem with types disappearing from the autocomplete?

File: react/src/AddDatasets/components/TableCells.tsx [hunk 1]
```diff
@@ -109,24 +105,54 @@ export function AutocompleteCell(
...
+        props.onEdit("");
+        //trigger dynamic option load
         if (props.onSearch) props.onSearch(value);
     };
 
-    const onEdit = (newValue: Option, autopopulate?: boolean) => {
-        onSearch(newValue.inputValue);
-        props.onEdit(newValue.inputValue, autopopulate);
-    };
+    const getIsFreeSolo = () => props.column.field === "notes";
```
Comment (kevinlul) [inline react/src/AddDatasets/components/TableCells.tsx:126]:
```suggestion
    const isFreeSolo = () => props.column.field === "notes";
```

**Review by kevinlul — CHANGES_REQUESTED** [review]

**Review by kevinlul — APPROVED** [review]
LGTM!

**Review by frewmack — APPROVED** [review]
LGTM!

**Pull request merged**